### PR TITLE
feat(thumbnail/basic): new thumbnail basic component

### PR DESCRIPTION
**Pull request by nguasch**
This PR creates the layout for this component:
screen shot 2017-07-17 at 13 22 44

Theme basic props defined here: https://github.com/SUI-Components/theme-basic/pull/73

### CODE REVIEW

File: components/thumbnail/basic/src/index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+
+class ThumbnailBasic extends Component {
+  render () {
+    const { image, captionText } = this.props
+    return (
+      <figure className='sui-ThumbnailBasic'>
+        <img className='sui-ThumbnailBasic-Image' src={image.src} alt={image.alt} />
```
Comment (JoanClaret) [inline components/thumbnail/basic/src/index.js]: `sui-ThumbnailBasic-Image` should be `sui-ThumbnailBasic-image`.
same for `sui-ThumbnailBasic-Caption` => `sui-ThumbnailBasic-caption`

more info https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md

File: components/thumbnail/basic/src/index.scss [hunk 2]
```diff
@@ -0,0 +1,44 @@
+@import '~@schibstedspain/theme-basic/lib/index';
+
+.sui-ThumbnailBasic {
+  @include reset-basic;
+  background: $bgc-thumbnail;
```
Comment (JoanClaret) [inline components/thumbnail/basic/src/index.scss]: in order to be more consistent with other components, i would use `$bgc-thumbnail-basic`. same for others selectors.

File: components/thumbnail/basic/src/index.scss [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    color: $c-thumbnail-caption;
+    display: block;
+    font-size: $fz-thumbnail-caption;
+    left: 0;
+    max-height: 0;
+    opacity: .9;
+    overflow: hidden;
+    position: absolute;
+    text-align: center;
+    transition: $transition-thumbnail;
```
Comment (JoanClaret) [inline components/thumbnail/basic/src/index.scss]: what about using `trs-thumbnail-basic`?
more info https://docs.emmet.io/cheat-sheet/

File: components/thumbnail/basic/src/index.scss [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      display: block;
+      padding: $p-thumbnail-caption;
+    }
+  }
+
+  &:hover {
+    .sui-ThumbnailBasic-Caption {
+      max-height: 100%;
+    }
+  }
```
Comment (JoanClaret) [inline components/thumbnail/basic/src/index.scss:43]: i'm sure there is a reason for this, but i don't understand what does this max-height on hover.
could you please shed light on this?

File: components/thumbnail/basic/src/index.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        </figcaption>
+      </figure>
+    )
+  }
+}
+
+ThumbnailBasic.displayName = 'ThumbnailBasic'
+
+ThumbnailBasic.propTypes = {
+  image: PropTypes.object.isRequired,
```
Comment (naxhh) [inline components/thumbnail/basic/src/index.js]: For better documentation would be a good idea to define the shape of the object

```
PropTypes.object.shape({
 src: PropTypes.string.isRequired,
 alt: PropTypes.string
})
```

File: components/thumbnail/basic/src/index.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  captionText: PropTypes.string
```
Comment (naxhh) [inline components/thumbnail/basic/src/index.js]: Keep in mind that you have set this to default, but in the render you seem to expect the `captionText` to exist.

Or this is required or you should check if the var exists before creating the figcaption.

Check a similar example:  https://github.com/SUI-Components/sui-components/blob/master/components/card/basic/src/index.js#L38

**Review by JoanClaret — CHANGES_REQUESTED** [review]
some minor comments, very good job anyways!
requesting changes only about the selector naming following SUIT naming convention.

File: demo/thumbnail/basic/context.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
```
Comment (naxhh) [inline demo/thumbnail/basic/context.js]: Why is this required?

**Comment by midudev** [comment]
Good job! It looks nice for me besides the comment written here! :)

File: components/thumbnail/basic/src/index.scss (hunk 3 above)
Comment (nguasch) [reply components/thumbnail/basic/src/index.scss]: 👍

File: components/thumbnail/basic/src/index.scss (hunk 2 above)
Comment (nguasch) [reply components/thumbnail/basic/src/index.scss]: 👍

File: components/thumbnail/basic/src/index.scss (hunk 4 above)
Comment (nguasch) [reply components/thumbnail/basic/src/index.scss:43]: It's because the transition. It's `max-height: 0` by default and goes to `max-height: 100%` on hover.

File: demo/thumbnail/basic/context.js (hunk 7 above)
Comment (jcger) [reply demo/thumbnail/basic/context.js]: It's autogenerated by sui-studio.

In case you already knew it: I don't want to check if every file that sui-studio generates makes sense or not, I assume that they do or at least that they would tell me that they are optional (like the props in the src/index.js file), therefore this file still exists.

**Review by carlosvillu — APPROVED** [review]
Great work!

File: components/thumbnail/basic/src/index.js (hunk 6 above)
Comment (jcger) [reply components/thumbnail/basic/src/index.js]: You'r right, after talking with Norbert, it makes no sense for us to use this component without caption so the caption will be required

File: components/thumbnail/basic/src/index.scss (hunk 4 above)
Comment (JoanClaret) [reply components/thumbnail/basic/src/index.scss:43]: aww, all right then! 👍

File: demo/thumbnail/basic/context.js (hunk 7 above)
Comment (naxhh) [reply demo/thumbnail/basic/context.js]: context is only created if you pass the flag `-C` when generating the files.
I was curious to know about this one since the component doesn't really need it because is not using anything from context.

Comment (jcger) [reply demo/thumbnail/basic/context.js]: Ok, will remove it

**Review by naxhh — APPROVED** [review]

File: components/thumbnail/basic/src/index.js (hunk 1 above)
Comment (nguasch) [reply components/thumbnail/basic/src/index.js]: Ok. I getting a little crazy with this convention 😅

File: components/thumbnail/basic/.gitignore [hunk 8]
```diff
@@ -0,0 +1,3 @@
+
+lib
+node_modules
```
Comment (davidbarna) [inline components/thumbnail/basic/.gitignore]: No need as it's already in root https://github.com/SUI-Components/sui-components/blob/feature/thumbnail-basic/.gitignore

Also, i would remove this file and add lib to root's .gitignore.

**Review by davidbarna — APPROVED** [review]

**Review by JoanClaret — APPROVED** [review]

File: components/thumbnail/basic/.gitignore (hunk 8 above)
Comment (jcger) [reply components/thumbnail/basic/.gitignore]: I think this is something enablers/code owners should talk about and solve properly for all components, for now this file is being autogenerated by sui-studio.

I can create a sui-studio issue if it's ok for you.

Comment (davidbarna) [reply components/thumbnail/basic/.gitignore]: "sui-studio generate" creates a general configuration. As sui-studio-create does not add gitignore, "sui-studio generate" can't assume you already added node_modules to the root folder.
But, if the root folder has a .gitignore, it makes sense you don't commit duplicated instruction in every single component.

**Review by davidbarna — APPROVED** [review]
